### PR TITLE
A new class for binning after combining 

### PR DESCRIPTION
**Pull request by nmahesh1412**
To give us the ability to binned individual days over GHA, we added:
1.) A class called CombinedBinnedData after CombinedData and before AverageData
2.) Also added steps in cli for the user to use the bin command for this step too

### CODE REVIEW

File: src/edges_analysis/analysis/levels.py [hunk 1]
```diff
@@ -468,7 +468,21 @@ def read_step(
     if fname.suffix == ".acq":
         return io.FieldSpectrum(fname).data
     else:
-        return _ReductionStep._get_object_class(fname)(fname)
+        return get_step_type(fname)(fname)
+
+
+def get_step_type(
+    fname: tp.PathLike | _ReductionStep | io.HDF5RawSpectrum,
+) -> _ReductionStep | io.HDF5RawSpectrum:
```
Comment (steven-murray) [inline src/edges_analysis/analysis/levels.py:476]: Ah, there should be a `Type[]` around the output types (i.e. `-> Type[_ReductionStep] | Type[io.HDF5RawSpectrum]`)

File: src/edges_analysis/analysis/levels.py [hunk 2]
```diff
@@ -2199,6 +2213,352 @@ def day_index(self, day: int) -> int:
         return self.days.tolist().index(day)
 
 
+@add_structure
+class CombinedBinnedData(_ModelMixin, _ReductionStep, _CombinedFileMixin):
+    """
+    Object representing gha binned data after observed days are combined.
+
+    Given a sequence of :class:`ModelData` objects, this class combines them into one
+    file, aligning them in (ideally small) bins in GHA/LST.
```
Comment (steven-murray) [inline src/edges_analysis/analysis/levels.py:2222]: Remove these lines

File: src/edges_analysis/analysis/levels.py [hunk 3]
```diff
@@ -2199,6 +2213,352 @@ def day_index(self, day: int) -> int:
...
+
+    See :class:`_ReductionStep` for documentation about the various datasets within this
+    class instance. Note that you can always check which data is inside each group by
+    checking its ``.keys()``.
+
+    See :meth:`CombinedData.promote` for detailed information about the processes
+    involved in creating this data from :class:`Level1` objects.
```
Comment (steven-murray) [inline src/edges_analysis/analysis/levels.py:2229]: Remove these lines

**Review by steven-murray — COMMENTED** [review]
Thanks @nmahesh1412 this is cool. Just a couple of small things to fix up.

File: src/edges_analysis/analysis/levels.py (hunk 1 above)
Comment (nmahesh1412) [reply src/edges_analysis/analysis/levels.py]: fixed

File: src/edges_analysis/analysis/levels.py (hunk 2 above)
Comment (nmahesh1412) [reply src/edges_analysis/analysis/levels.py]: done

File: src/edges_analysis/analysis/levels.py (hunk 3 above)
Comment (nmahesh1412) [reply src/edges_analysis/analysis/levels.py]: done

**Comment by steven-murray** [comment]
Ah. OK one thing we need now: tests!

File: src/edges_analysis/analysis/levels.py [hunk 4]
```diff
@@ -1044,51 +1046,32 @@ def _promote(
...
-            An instantiated :class:`CalibratedData` object.
         """
         t = time.time()
-        q = prev_step["spectra"]["Q"]
+        freq_f = prev_step["freq_ancillary"]["frequencies"]
+        f_mask = np.where(freq_f.freq >= 40)[0]
+        freq = freq_f[f_mask:]
+        q = prev_step["spectra"]["Q"][
+            f_mask:,
+        ]
```
Comment (steven-murray) [inline src/edges_analysis/analysis/levels.py:1062]: I think a quicker/neater way of doing the masking here would be to use the `FrequencyRange` object:

```python
freq_rng = FrequencyRange(prev_step['freq_ancillary']['frequencies'], f_low=40.0)
freq = freq_rng.freq
q = prev_step['spectra']['Q'][freq_rng.mask]
...
```

**Review by steven-murray — APPROVED** [review]
I think this is good, pending tests passing etc. Thanks @nmahesh1412 !

**Pull request merged**